src/hooks/useBalanceChartData/useBalanceChartData.ts

```diff
@@ -234,27 +247,30 @@ export const calculateBucketPrices: CalculateBucketPrices = args => {
         }
       }
 
+      // Identify Special cases where we should not include cosmos delegate/undelegate txs in chart balance
+      const skipBucketBalanceChange = skipCosmosTx(tx)
+
       tx.transfers.forEach(transfer => {
         const asset = transfer.caip19
 
         if (!assetIds.includes(asset)) return
 
         const bucketValue = bnOrZero(bucket.balance.crypto[asset])
         const transferValue = bnOrZero(transfer.value)
-
-        switch (transfer.type) {
-          case chainAdapters.TxType.Send:
-            // we're going backwards, so a send means we had more before
-            bucket.balance.crypto[asset] = bucketValue.plus(transferValue)
-            break
-          case chainAdapters.TxType.Receive:
-            // we're going backwards, so a receive means we had less before
-            bucket.balance.crypto[asset] = bucketValue.minus(transferValue)
-            break
-          default: {
-            console.warn(`calculateBucketPrices: unknown tx type ${transfer.type}`)
+        if (!skipBucketBalanceChange)
```

Comment:
do this on a single line after 256 and just return, don't make a christmas tree unnecessarily